qutebrowser/mainwindow/tabwidget.py

```diff
@@ -479,14 +483,16 @@ def tabSizeHint(self, index):
                     return size
 
             # If we *do* have enough space, tabs should occupy the whole window
-            # width.
-            #looks like this generates high cpu usage
-            #need to register the number of pin tabs in advance
-            #nb_of_pins = len([None for item in range(self.count()) 
-            #                    if objreg.get('tab', scope='tab',
-            #                            window=self._win_id, tab=item).pin is True])
-            #width = (self.width() + 40*nb_of_pins) / self.count()
-            width = self.width() / self.count()
+            # width. Also taken in consideration the reduced space necessary for
+            # the pinned tabs.
+            #TODO: During shutdown the self.count goes down, but the self.pinned not
+            #this generates some odd bahavior.
+            #To avoid this we compare self.count against self.pinned.
+            if self.pinned > 0 and self.count() > self.pinned:
+                width = (self.width() - 40*self.pinned) / (self.count() - self.pinned)
```

Comment:
Couldn't you use [tabSizeHint](http://doc.qt.io/qt-5/qtabbar.html#tabSizeHint) to get the space needed for pinned tabs exactly instead of hardcoding it here?